bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/GardenaSmartImpl.java

```diff
@@ -153,9 +155,14 @@ public GardenaSmartImpl(String id, GardenaConfig config, GardenaSmartEventListen
     private void startWebsockets() throws Exception {
         for (LocationDataItem location : locationsResponse.data) {
             WebSocketCreatedResponse webSocketCreatedResponse = getWebsocketInfo(location.id);
-            String socketId = id + "-" + location.attributes.name;
+            Location locationAttributes = location.attributes;
```

Comment:
> this would IMHO trigger even more refactoring not originally planned for avoiding the compiler warning: Renaming the DTO's.

Not at all. I am not suggesting renaming anything IN the DTO's; and specifically not suggesting the generic T `attributes` field. I am suggesting renaming the specific local variables that you now have to create when you cast the generic T `attributes` field locally to a reference to a specific class type.

I half understand your reluctance about losing the word 'attributes' from such local variable names. But then I would suggest that you use the name 'locationAttributes' / 'webSocketAttributes' / etc. for any case where a a generic attributes is cast to a specific class type i.e. use 'specificClassNameAttributes' rather than just 'attributes'..

Reply:
I understand. My point was that to avoid misunderstanding from that original suggested renaming, probably it would need some more refactoring, for example to avoid confusion between **socket** of type `GardenaWebSocket` vs. **webSocket** of type `WebSocket`.

I have renamed as suggested now to have full prefix of type in variable names.